lib/active_merchant/billing/gateways/xpay.rb

```diff
@@ -162,9 +176,17 @@ def add_exemptions(post, options)
         post[:exemptions] = options[:exemptions] || 'NO_PREFERENCE'
       end
 
-      def add_3ds_params(post, options)
-        post[:threeDSAuthData] = { threeDSAuthResponse: options[:three_ds_auth_response] }.compact
-        post[:operationId] = options[:operation_id] if options[:operation_id]
+      def add_3ds_params(post, validation)
```

Comment:
💬 Comment / Style:

Not a big deal just a little more Dryer:

```ruby
def add_3ds_params(post, validation)
  post[:threeDSAuthData] = {
    authenticationValue: validation['threeDSAuthResult']['authenticationValue'],
    eci: validation['threeDSAuthResult']['eci'],
    xid: validation['threeDSAuthResult']['xid']
  }
  post[:operationId] = validation['operation']['operationId']
end
```

Reply:
Done!